packages/components/src/spectrum/View.tsx

```diff
@@ -43,34 +52,48 @@ export const View = forwardRef<DOMRefValue<HTMLElement>, ViewProps>(
       ...rest
     } = props;
 
-    // Using shorthand to define border styling. Removing a style property during rerender can lead to styling bugs
-    // ex. changing from borderXColor = blue to borderColor = blue will cause the left and right borders to disappear
-    const defaultBorderColor = colorValueStyle(borderColor) ?? 'transparent';
-    const defaultBorderXColor =
-      colorValueStyle(borderXColor) ?? defaultBorderColor;
-    const defaultBorderYColor =
-      colorValueStyle(borderYColor) ?? defaultBorderColor;
-    const topColor = colorValueStyle(borderTopColor) ?? defaultBorderYColor;
-    const bottomColor =
-      colorValueStyle(borderBottomColor) ?? defaultBorderYColor;
-    const leftColor = colorValueStyle(borderStartColor) ?? defaultBorderXColor;
-    const rightColor = colorValueStyle(borderEndColor) ?? defaultBorderXColor;
+    const style = useMemo(() => {
+      const borderStyle: CSSProperties = {};
+      if (borderColor !== undefined) {
+        borderStyle.borderColor = colorValueStyle(borderColor);
```

Comment:
It would be nice if we could just pass our own conversion function into `View`... Right now `View` is using `useStyleProps` with the `viewStyleProps` handlers to convert functions: https://github.com/adobe/react-spectrum/blob/bd458c1ed166a5ff1fd93f0c1a397e1540b3d880/packages/%40react-spectrum/view/src/View.tsx#L26C43-L26C57
If we could just pass our own color conversion handler in there somehow, then we wouldn't have to rewrite these props for every color: https://github.com/adobe/react-spectrum/blob/bd458c1ed166a5ff1fd93f0c1a397e1540b3d880/packages/%40react-spectrum/utils/src/styleProps.ts#L263
I'm tempted to file a ticket/PR in the Adobe GitHub repo to add this functionality, since we're ending up having to duplicate a bunch of things here just to handle the borders... but this should be fine for now.